ui/app/AppLayouts/Onboarding/shared/CreatePasswordModal.qml

```diff
@@ -28,66 +29,122 @@ ModalPopup {
             }
         }
-        onTextChanged: {
-            [repeatPasswordFieldValid, repeatPasswordValidationError] =
-                Utils.validatePasswords("repeat", firstPasswordField, repeatPasswordField);
+
+        Input {
+            id: repeatPasswordField
+            visible: !userProfile.isKeycardUser
+            width: parent.width - 2 * d.padding
+            enabled: firstPasswordFieldValid
+            placeholderText: qsTr("Confirm password…")
+            textField.echoMode: TextInput.Password
+            Keys.onReturnPressed: function(event) {
+                if (submitBtn.enabled) {
+                    submitBtn.clicked(event)
+                }
+            }
+            onTextChanged: {
+                [repeatPasswordFieldValid, repeatPasswordValidationError] =
+                                                                          Utils.validatePasswords("repeat", firstPasswordField, repeatPasswordField);
+            }
         }
-    }
 
-    StyledText {
-        id: validationError
-        text: {
-            if (passwordValidationError !== "") return passwordValidationError;
-            if (repeatPasswordValidationError !== "") return repeatPasswordValidationError;
-            return "";
+        StyledText {
+            anchors.horizontalCenter: parent.horizontalCenter
+            visible: userProfile.isKeycardUser
+            text: qsTr("Enter new PIN")
+            font.pixelSize: d.fontSize
+        }
+
+        StatusPinInput {
+            id: firstPinInputField
+            anchors.horizontalCenter: parent.horizontalCenter
+            visible: userProfile.isKeycardUser
+            validator: StatusIntValidator{bottom: 0; top: 999999;}
+            pinLen: Constants.keycard.general.keycardPinLength
+
+            onPinInputChanged: {
+                if (pinInput.length == Constants.keycard.general.keycardPinLength) {
+                    repeatPinInputField.statesInitialization()
+                    repeatPinInputField.forceFocus()
+                }
+
+                [firstPasswordFieldValid, passwordValidationError] =
+                                                                   Utils.validatePINs("first", firstPinInputField, repeatPinInputField);
+                [repeatPasswordFieldValid, repeatPasswordValidationError] =
+                                                                          Utils.validatePINs("repeat", firstPinInputField, repeatPinInputField);
+            }
+        }
+
+        StyledText {
+            anchors.horizontalCenter: parent.horizontalCenter
+            visible: userProfile.isKeycardUser
+            text: qsTr("Repeat PIN")
+            font.pixelSize: d.fontSize
+        }
+
+        StatusPinInput {
+            id: repeatPinInputField
+            anchors.horizontalCenter: parent.horizontalCenter
+            visible: userProfile.isKeycardUser
+            validator: StatusIntValidator{bottom: 0; top: 999999;}
```

Comment:
Number.MAX_SAFE_INTEGER

Reply:
@MishkaRogachev cannot use max int here, cause it validates up to 6 digits pin. So pin may be any number form `000000` to `999999`.